File: Recursion/digits_of_number.c
#include <stdio.h>

void printDigits(int n) {
    if (n < 10) {
        printf("%d ", n);
        return;
    }
    printDigits(n / 10);
    printf("%d ", n % 10);
}

int main() {
    int number;
    printf("Enter a number: ");
    scanf("%d", &number);
    printf("Digits of the number: ");
    printDigits(number);
    printf("\n");
    return 0;
}

File: Recursion/digits_of_number.py
def print_digits(n):
    if n < 10:
        print(n, end=" ")
        return
    print_digits(n // 10)
    print(n % 10, end=" ")

number = int(input("Enter a number: "))
print("Digits of the number:", end=" ")
print_digits(number)
print()
